infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutSearchService.kt

```diff
@@ -22,48 +22,127 @@ class LayoutSearchService @Autowired constructor(
             "searchAssets",
             PUBLICATION_STATE to publicationState,
             "searchTerm" to searchTerm,
             "limitPerResultType" to limitPerResultType,
-            "contextLocationTrackId" to contextLocationTrackId,
+            "locationTrackSearchScope" to locationTrackSearchScope,
         )
 
-        return if (contextLocationTrackId != null) {
-            val switches = locationTrackService.getSwitchesForLocationTrack(contextLocationTrackId, publicationState)
-                .let { ids -> switchService.getMany(publicationState, ids) }
-            val locationTracks = locationTrackService
-                .getWithAlignmentOrThrow(publicationState, contextLocationTrackId)
-                .let { (lt, alignment) ->
-                    val duplicates = locationTrackService
-                        .getLocationTrackDuplicates(lt, alignment, publicationState)
-                        .map { it.id }
-                        .let { ids -> locationTrackService.getMany(publicationState, ids) }
+        return if (locationTrackSearchScope != null) {
+            searchByLocationTrackSearchScope(locationTrackSearchScope, publicationState, searchTerm, limitPerResultType)
+        } else {
+            searchFromEntireRailwayNetwork(publicationState, searchTerm, limitPerResultType)
+        }
+    }
 
-                    listOf(lt) + duplicates
-                }
-            val trackNumbers = emptyList<TrackLayoutTrackNumber>()
+    fun searchAllLocationTracks(
+        publicationState: PublicationState,
+        searchTerm: FreeText,
+        limit: Int,
+    ): List<LocationTrack> {
+        logger.serviceCall(
+            "searchAllLocationTracks",
+            "publicationState" to publicationState,
+            "searchTerm" to searchTerm,
+            "limit" to limit,
+        )
 
-            TrackLayoutSearchResult(
-                switches = switches
-                    .let { list -> switchService.filterBySearchTerm(list, searchTerm) }
-                    .take(limitPerResultType),
-                locationTracks = locationTracks
-                    .let { list -> locationTrackService.filterBySearchTerm(list, searchTerm) }
-                    .take(limitPerResultType),
-                trackNumbers = trackNumbers
-                    .let { list -> trackNumberService.filterBySearchTerm(list, searchTerm) }
-                    .take(limitPerResultType)
-            )
-        } else {
-            TrackLayoutSearchResult(
-                switches = switchService.list(publicationState, searchTerm, limitPerResultType),
-                locationTracks = locationTrackService.list(publicationState, searchTerm, limitPerResultType),
-                trackNumbers = trackNumberService.list(publicationState, searchTerm, limitPerResultType),
+        return locationTrackService.list(publicationState, true)
+            .let { list -> locationTrackService.filterBySearchTerm(list, searchTerm) }
```

Comment:
En löytänyt nopeaa tapaa siirtää näitä tänne, joten ne pysyvät edelleen assettityyppikohtaisissa serviceissä. Ei sekään olisi ehkä maailman valtavin jumppa, mutta tein kuitenkin rajanvedon tähän ettei nämä reviewkorjaukset pullataikinoidu ihan täysin.

Reply:
Joo. Se on vähän kiemura rakenne. Tähän search serviceen varmasti tehtävissä että siirtäisi senkin, mutta sen voi katsoa myöhemminkin.